tools/clang/test/HLSL/intrinsic-examples.hlsl

```diff
@@ -32,27 +32,24 @@ float4 RWByteAddressBufferMain(uint2 a : A, uint2 b : B) : SV_Target
-  r += uav1.Load2<float>(16);                               /* expected-error {{Unrecognized template arguments for intrinsic Load2}} */
-  r += uav1.Load3<int>(20);                                 /* expected-error {{Unrecognized template arguments for intrinsic Load3}} */
-  r += uav1.Load4<int16_t>(24);                             /* expected-error {{Unrecognized template arguments for intrinsic Load4}} */
-  r += uav1.Load<half3x4>(24);                              /* expected-error {{Unrecognized template arguments for intrinsic Load}} expected-error {{cannot convert from 'matrix<__fp16, 3, 4>' to 'float4'}} */
+  r += uav1.Load<float, float3>(16);                        /* expected-error {{Unrecognized template arguments for intrinsic Load, One scalar or vector type argument up to 16 bytes in size supported.}} */
+  r += uav1.Load<double3>(16);                              /* expected-error {{Unrecognized template arguments for intrinsic Load, One scalar or vector type argument up to 16 bytes in size supported.}} expected-error {{cannot convert from 'vector<double, 3>' to 'float4'}} */
+  r += uav1.Load2<float>(16);                               /* expected-error {{Unrecognized template arguments for intrinsic Load2, No template arguments supported.}} */
+  r += uav1.Load3<int>(20);                                 /* expected-error {{Unrecognized template arguments for intrinsic Load3, No template arguments supported.}} */
+  r += uav1.Load4<int16_t>(24);                                 /* expected-error {{Unrecognized template arguments for intrinsic Load4, No template arguments supported.}} */
+  r += uav1.Load<half3x4>(24);                              /* expected-error {{Unrecognized template arguments for intrinsic Load, One scalar or vector type argument up to 16 bytes in size supported.}} expected-error {{cannot convert from 'matrix<__fp16, 3, 4>' to 'float4'}} */
   // valid template argument
   uav1.Store(0, r);
-  uav1.Store<float>(0, r.x);
-  uav1.Store<float1>(0, r.x);
-  uav1.Store<half2>(0, r.xy);                               /* expected-warning {{conversion from larger type 'vector<float, 2>' to smaller type 'half2', possible loss of data}} */
-  uav1.Store<uint3>(0, r.xyz);
-  uav1.Store<float4>(0, r);
-  uav1.Store<half4>(0, r);                                  /* expected-warning {{conversion from larger type 'float4' to smaller type 'half4', possible loss of data}} */
-  // warnings
-  uav1.Store<float2>(0, r);                                 /* expected-warning {{implicit truncation of vector type}} */
-  uav1.Store<half3>(0, r);                                  /* expected-warning {{conversion from larger type 'float4' to smaller type 'half3', possible loss of data}} expected-warning {{implicit truncation of vector type}} */
+  uav1.Store(0, r.x);
+  uav1.Store(0, (half2)r.xy);
+  uav1.Store(0, (int3)r.xyz);
+  uav1.Store(0, (double2)r.xy);
   // errors
```

Comment:
I think we need at least a `uav1.Store(0, (double3) r.xyz);` test for that error case.